File: backend/src/main/java/com/g0urd/grad_project/service/DriverService/DriverService.java
package com.g0urd.grad_project.service.DriverService;

import java.util.List;

import com.g0urd.grad_project.models.Driver;

public interface DriverService {

    List<Driver> fetchAllDrivers();

    Driver findById(Long id);

    Driver createDriver(Driver driver);

    Driver updateDriver(Driver driver);

    Boolean deleteDriver(Long id);

}

File: frontend/src/app/service/driver/driver.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Driver } from '../../component/driver/driver';
import { Observable } from 'rxjs';

const BASE_URL = ["http://localhost:8080/api/scheduler"]

@Injectable({
  providedIn: 'root'
})
export class DriverService {

  constructor(private http: HttpClient) { }

  fetchAllDrivers():Observable<Driver[]> {
    return this.http.get<Driver[]>(BASE_URL + `/driver`);
  }

  createDriver(data: Driver) {
    return this.http.post<Driver>(BASE_URL + `/driver`, data);
  }

  updateDriver(data: Driver) {
    return this.http.put<Driver>(BASE_URL + `/driver/${data.id_driver}`, data);
  }

  deleteDriver(id_driver: Number) {
    return this.http.delete<Driver>(BASE_URL + `/driver/${id_driver}`);
  }
}
